ibis/backends/base/sql/alchemy/registry.py

```diff
@@ -541,7 +541,9 @@ def translate(t, expr):
     ops.BitOr: reduction(sa.func.bit_or),
     ops.BitXor: reduction(sa.func.bit_xor),
     ops.CountDistinct: reduction(lambda arg: sa.func.count(arg.distinct())),
-    ops.HLLCardinality: reduction(lambda arg: sa.func.count(arg.distinct())),
```

Comment:
Let's leave the duplicated rules in until we delete `HLLCardinality`/`CMSMedian` in case anyone is constructing these operations externally. We'll delete the duplicated rules when we delete the classes.

Reply:
I did find this: https://ibis-project.org/docs/3.0.2/user_guide/design/

Reply:
I am going to attempt to do this, but could you teach me what is meant by rules in this context? The registry lists in the `registry.py`s? Are the `@execute_node.register()` also creating rules?

Reply:
Sure!

I'm using the word "rule" here informally to mean a mapping from an `ibis.expr.operations.Node` subclass to a function that produces something in the backend's target representation (SQL string, SQLAlchemy expression, Dask DAG, etc.).

`@execute_node.register` **as used in the pandas backend** doesn't quite fit into that because there's no target representation. In that case the rule is mapping an operation to code that will execute the operation directly and return a DataFrame/Series/scalar.

Does that help?